python_modules/dagster/dagster_tests/core_tests/test_configured.py

```diff
@@ -35,3 +55,156 @@ def mypipeline():
     result = execute_pipeline(mypipeline)
 
     assert result.success
+
+
+get_asset_keys_from_materializations = lambda materializations: sorted(
+    [materialization.asset_key for materialization in materializations]
+)
+
+
+def test_configured_asset():
+    @asset(config_schema={"greeting": str}, key_prefix=["prefix"], group_name="my_group")
+    def my_asset(context):
+        return context.op_config["greeting"]
+
+    configured_hello = my_asset.configured({"greeting": "hello"}, "configured_hello")
+    assert configured_hello.group_names_by_key == {
+        AssetKey(["prefix", "configured_hello"]): "my_group"
+    }
+    result = materialize([configured_hello])
+    assert result.success
+    assert result.output_for_node("configured_hello") == "hello"
+
+    configured_hi = my_asset.configured({"greeting": "hi"}, "configured_hi")
+    result = materialize([configured_hi])
+    assert result.success
+    assert result.output_for_node("configured_hi") == "hi"
+
+    test_job = define_asset_job("test_job").resolve([configured_hello, configured_hi, my_asset], [])
+    result = test_job.execute_in_process(
+        run_config={"ops": {"prefix__my_asset": {"config": {"greeting": "hey"}}}}
+    )
+    assert result.success
+    assert result.output_for_node("prefix__my_asset") == "hey"
+    assert result.output_for_node("configured_hello") == "hello"
+    assert result.output_for_node("configured_hi") == "hi"
+    assert get_asset_keys_from_materializations(
+        result.asset_materializations_for_node("prefix__my_asset")
+    ) == [AssetKey(["prefix", "my_asset"])]
+    assert get_asset_keys_from_materializations(
+        result.asset_materializations_for_node("configured_hello")
+    ) == [AssetKey(["prefix", "configured_hello"])]
+    assert get_asset_keys_from_materializations(
+        result.asset_materializations_for_node("configured_hi")
+    ) == [AssetKey(["prefix", "configured_hi"])]
+
+
```

Comment:
what happens when an asset is constructed without a decorator (assuming that is a pattern we support)? As mentioned in another comment, I think you can construct some pathological cases here where it's not sufficient to assume that the name / asset key are equivalent.